authn/authn.go

```diff
@@ -43,55 +50,79 @@ func NewClient(config Config) (*Client, error) {
 // ID and should be used as a unique foreign key in your users data.
 //
 // If the JWT does not verify, the returned error will explain why. This is for debugging purposes.
-func (ac *Client) SubjectFrom(idToken string) (string, error) {
-	claims, err := ac.verifier.GetVerifiedClaims(idToken)
+// It is also possible to overwrite the default audience used to verify the JWT by passing
+// a jwt.Audience{} as the second parameter.
+func (ac *Client) SubjectFrom(idToken string, options ...interface{}) (string, error) {
```

Comment:
I'm divided between variadic options and `SubjectFromWithAudience`. The variadic options don't document themselves very well, they enforce arguments at runtime, and the options parsing promises to grow exponentially in complexity. But they're more stable in case of unexpected additions.

Do you have opinions from your own experience?

I'm starting to lean towards attempting `SubjectFromWithAudience` now, deferring variadic options until `SubjectFrom` has a second option (if ever).

Reply:
Yeah I also wasn't sure because it feels strange to accept variadic interface args. I normally go with an interface type and "option functions" like this:
```golang
type Option func(v *verifier) error

func WithAudience(audience jwt.Audience) Option {
    return func(v *verifier) error {
        v.audience = audience
        return nil
    }
}

func (ac *Client) SubjectFrom(token stirng, options ...Option) (int, error) {
    // ...

   for _, opt := range options {
       if err := opt(verifier); err != nil {
           return -1, err
       }
    }

    // ...
}
```
and use it like `client.SubjectFrom(token, authn.WithAudience(myAudience))`. But I'm unsure this is worth the effort if `SubjectFrom` will never receive a second option. In addition, the changes I proposed are not actually backwards compatible because in one of my own projects I used
```golang
type SubjectExtractor func(idToken string) (string, error)
// ...
var extractor SubjectExtractor = authnClient.SubjectFrom
```
which needed to be changed to the new function signature. So I guess a `SubjectFromWithAudience` will be the best option for now.

Reply:
Since a new function will serve the purpose of this change best I reverted `SubjectFrom` to it's previous signature and added `SubjectFromWithAudience`